src/Compiler/Backend/GLSL/GLSLConverter.cpp

```diff
@@ -813,6 +813,9 @@ void GLSLConverter::ConvertIntrinsicCall(CallExpr* ast)
         case Intrinsic::Texture_SampleLevel_5:
             ConvertIntrinsicCallTextureSampleLevel(ast);
             break;
+        case Intrinsic::Image_Store:
```

Comment:
All the other cast related code seem to be there so I added it there as it seemed appropriate. But other than that no specific reason.